### DISCUSSION TITLE
Async `eventContent` function

### DISCUSSION THREAD

**Discussion by lukas-frey** [vkurko/calendar#292] [Q&A] [answered]
Hi, is it possible to make the `eventContent` async, so I can process a Promise within the function to load the content of an event from an API?

I'm currently struggling with making it work, as soon as I make the `eventContent` function async, the calendar doesn't render properly.

It just appears for a second with the correct data and then disappears.

**Comment by vkurko** [comment]
> is it possible to make the `eventContent` async

Now this is not possible. But perhaps in your case it is better to load the content in a different way, for example, via `eventSources`. Can you provide more details on how your implementation works?

**Reply by lukas-frey** [reply]
> > is it possible to make the `eventContent` async
> 
> Now this is not possible. But perhaps in your case it is better to load the content in a different way, for example, via `eventSources`. Can you provide more details on how your implementation works?

Are you able to define the HTML per-event when using eventSources?

Depending on which user is currently viewing the calendar, I need to show different type of data from the extendedProps of an event.

**Reply by vkurko** [reply, marked as answer]
Can you load all the required data from the server via `eventSources`? Then you won't need to load anything extra in the `eventContent` callback.

**Reply by lukas-frey** [reply]
I see, I guess that could work - loading all data and conditionally hiding those that shouldn't be visible at certain times. 🤔 I'll try that then, thanks.

**Reply by vkurko** [reply]
You may not load all the data, but only those that can actually be shown to the current user. The question is where exactly the loading should take place. Not inside `eventContent`, but in `eventSources`. To reload the data (if, for example, the user has changed) you can use `refetchEvents` method.
